3scale/porta
# Support OAS3 in Admin Portal (ActiveDocs)

> ⚠️ Merge only after #1596

**What this PR does / why we need it**:

Introduces `swagger-ui-react` in ActiveDocs in order to support OAS3 / Swagger 3:

![Screen Shot 2020-01-16 at 12 40 52](https://user-images.githubusercontent.com/11672286/72712704-db007080-3b6b-11ea-8093-aac258711868.png)

![Screen Shot 2020-01-28 at 10 48 05](https://user-images.githubusercontent.com/11672286/73253529-fea36680-41bc-11ea-94ff-d288b642f4d8.png)


**Which issue(s) this PR fixes** 

[THREESCALE-2725: Start using SwaggerUI 3.18.3 or higher for OAS3 specs](https://issues.redhat.com/browse/THREESCALE-2725)

**Verification steps** 
1. Go to any Product's active docs section: apiconfig/services/2/api_docs/<id>/preview#/
2. Create a new spec (v3) i.e. [Swagger petstore](http://petstore.swagger.io:8080/api/v3/openapi.json)
3. It should display the new UI
4. Opening a spec (v2) should still display the old UI:
    ![Screen Shot 2020-01-28 at 10 57 47](https://user-images.githubusercontent.com/11672286/73253559-08c56500-41bd-11ea-8337-b0b4fe95f282.png)


**Special notes for your reviewer**:
Dev portal support and proper styling will be handled in a following PR (#1595).
